### PR TITLE
delete Evicted pods first during pod gc

### PR DESCRIPTION
**Pull request by borgerli**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

When gc-ing pods, kcm only cares about creation time and pod name. In a cluster with losts of Evicted pods and otehr kind of terminated pods, such as created by job controllers, controller-created pods may be deleted first, which will impact on controller behaviours. 
With this PR, Evicted pods will be deleted first, and try to avoid delete other terminated pods as best as possible.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
When GCing pods, kube-controller-manager will delete Evicted pods first.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by aojea** [comment]
@wojtek-t who may own this?

**Review by wojtek-t — COMMENTED** [review]
I have one small comment - other than that LGTM.

File: pkg/controller/podgc/gc_controller.go [hunk 1]
```diff
@@ -308,13 +309,22 @@ func (gcc *PodGCController) gcUnscheduledTerminating(ctx context.Context, pods [
...
+type byEvictionAndCreationTimestamp []*v1.Pod
 
-func (o byCreationTimestamp) Len() int      { return len(o) }
-func (o byCreationTimestamp) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
+func (o byEvictionAndCreationTimestamp) Len() int      { return len(o) }
+func (o byEvictionAndCreationTimestamp) Swap(i, j int) { o[i], o[j] = o[j], o[i] }
 
-func (o byCreationTimestamp) Less(i, j int) bool {
+func (o byEvictionAndCreationTimestamp) Less(i, j int) bool {
+	if o[i].Status.Reason != o[j].Status.Reason {
```
[the remark this answers is not in the document]
Comment (borgerli) [reply pkg/controller/podgc/gc_controller.go]: @wojtek-t Thanks for the great point. I updated code as suggested.

**Pull request merged**